django/db/models/query.py

```diff
@@ -1907,6 +1908,9 @@ def _clone(self):
     def _fetch_all(self):
         if self._result_cache is None:
             self._result_cache = list(self._iterable_class(self))
+            if settings.DEBUG:
+                self.executed_query = connections[self.db].queries[-1]["sql"]
```

Comment:
A debug cursor [can also be forced](https://github.com/django/django/blob/bc9b6251e0b54c3b5520e3c66578041cc17e4a28/django/db/backends/base/base.py#L171)

```suggestion
            self.executed_query = connections[self.db].last_query
```